dexlib2/src/main/java/org/jf/dexlib2/dexbacked/DexBackedMethodImplementation.java

```diff
@@ -148,4 +149,35 @@ private DebugInfo getDebugInfo() {
     public Iterator<String> getParameterNames(@Nullable DexReader dexReader) {
         return getDebugInfo().getParameterNames(dexReader);
     }
+
+    /**
+     * Calculate and return the private size of a method implementation.
+     *
+     * Calculated as: debug info size + instructions size + try-catch size
+     *
+     * @return size in bytes
+     */
+    public int getSize(){
+        int debugSize = getDebugInfo().getSize();
+
+        //move to the end of instructions list and read offset
+        Iterator<? extends Instruction> iter = getInstructions().iterator();
```

Comment:
You should be able to directly skip over the instructions, using the code_item.insns_size field, instead of iterating over them.